src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -876,7 +876,10 @@ public function addQuantityInformation(
         $show_price = $this->shouldShowPrice($settings, $product);
         $show_availability = $show_price && $settings->stock_management_enabled;
         $this->product['show_availability'] = $show_availability;
-        $product['quantity_wanted'] = $this->getQuantityWanted();
+
+        if (empty($product['quantity_wanted'])) {
```

Comment:
```suggestion
        if (!isset($product['quantity_wanted'])) {
```
Otherwise if it's already 0 it will call the method for nothing, no?

Reply:
Is there any scenario where someone wants to pass quantity_wanted as 0? :) I don't think of any to be honest

Reply:
It's a ProductPresenter, maybe it's in order to display products, and for optimization you can directly set the quantity_wanted at 0 to prevent this call.

Reply:
@PierreRambaud  let's use `isset`, just to make sure 💪